components/director/internal/domain/api/service_test.go

```diff
@@ -1 +1,264 @@
+		},
+		DefaultAuth: &model.Auth{},
+		Version:     &model.Version{},
+	}
+
+	ctx := context.TODO()
+	ctx = tenant.SaveToContext(ctx, "tenant")
+
+	testCases := []struct {
+		Name               string
+		RepositoryFn       func() *automock.APIRepository
+		Input              model.APIDefinitionInput
+		InputID            string
+		ExpectedErrMessage string
+	}{
+		{
+			Name: "Success",
+			RepositoryFn: func() *automock.APIRepository {
+				repo := &automock.APIRepository{}
+				repo.On("GetByID", "foo").Return(apiDefinitionModel, nil).Once()
+				repo.On("Update", inputApiDefinitionModel).Return(nil).Once()
+				return repo
+			},
+			InputID:            "foo",
+			Input:              modelInput,
+			ExpectedErrMessage: "",
+		},
+		{
+			Name: "Update Error",
+			RepositoryFn: func() *automock.APIRepository {
+				repo := &automock.APIRepository{}
+				repo.On("GetByID", "foo").Return(apiDefinitionModel, nil).Once()
+				repo.On("Update", inputApiDefinitionModel).Return(testErr).Once()
+				return repo
+			},
+			InputID:            "foo",
+			Input:              modelInput,
+			ExpectedErrMessage: testErr.Error(),
+		},
+		{
+			Name: "Get Error",
+			RepositoryFn: func() *automock.APIRepository {
+				repo := &automock.APIRepository{}
+				repo.On("GetByID", "foo").Return(nil, testErr).Once()
+				return repo
+			},
+			InputID:            "foo",
+			Input:              modelInput,
+			ExpectedErrMessage: testErr.Error(),
+		},
+	}
+
+	for i, testCase := range testCases {
+		t.Run(fmt.Sprintf("%d: %s", i, testCase.Name), func(t *testing.T) {
+			repo := testCase.RepositoryFn()
+
+			svc := api.NewService(repo)
+
+			// when
+			err := svc.Update(ctx, testCase.InputID, testCase.Input)
+
+			// then
+			if testCase.ExpectedErrMessage == "" {
+				require.NoError(t, err)
+			} else {
+				assert.Contains(t, err.Error(), testCase.ExpectedErrMessage)
+			}
+
+			repo.AssertExpectations(t)
+		})
+	}
+}
+
+func TestService_Delete(t *testing.T) {
+	// given
+	testErr := errors.New("Test error")
+
+	id := "foo"
+
+	apiDefinitionModel := &model.APIDefinition{
+		Name:          "Bar",
+		ApplicationID: "id",
+		TargetURL:     "https://test-url-updated.com",
+		Spec: &model.APISpec{
+			FetchRequest: &model.FetchRequest{
+				Mode: model.FetchModePackage,
+			},
+		},
+		Version: &model.Version{},
+	}
+
+	ctx := context.TODO()
+	ctx = tenant.SaveToContext(ctx, "tenant")
+
+	testCases := []struct {
+		Name               string
+		RepositoryFn       func() *automock.APIRepository
+		Input              model.APIDefinitionInput
+		InputID            string
+		ExpectedErrMessage string
+	}{
+		{
+			Name: "Success",
+			RepositoryFn: func() *automock.APIRepository {
+				repo := &automock.APIRepository{}
+				repo.On("GetByID", id).Return(apiDefinitionModel, nil).Once()
+				repo.On("Delete", apiDefinitionModel).Return(nil).Once()
+				return repo
+			},
+			InputID:            id,
+			ExpectedErrMessage: "",
+		},
+		{
+			Name: "Delete Error",
```

Comment:
in other places you have more descriptive test names which is better IMO

Reply:
It's not that good, because `go test` adds `_` between spaces while running tests, so longer name is more difficult to read